controllers/repository_controller.go

```diff
@@ -161,7 +161,14 @@ func (r *RepositoryReconciler) UpdateRepository(ctx context.Context, logger logr
 		}
 		return false, err
 	}
-
+	if v, ok := instanceDeepCopy.Labels[corev1alpha1.RepositorySourceLabel]; !ok || (v != string(corev1alpha1.Opensource) && v != string(corev1alpha1.Official) && v != string(corev1alpha1.Normal)) {
```

Comment:
There are two conditions for adding a label here, I think we can use a variable to determine if the instance needs to be updated, so that these two judgment conditions will only generate one update event, otherwise it will generate two update events.